src/equations/compressible_euler_potential_temperature_2d.jl

```diff
@@ -0,0 +1,301 @@
+                                              equations::CompressibleEulerPotentialTemperatureEquations2D)
+    # flip sign of normal to make it outward pointing, then flip the sign of the normal flux back
+    # to be inward pointing on the -x and -y sides due to the orientation convention used by StructuredMesh
+    if isodd(direction)
+        boundary_flux = -boundary_condition_slip_wall(u_inner, -normal_direction,
+                                                      x, t, surface_flux_function,
+                                                      equations)
+    else
+        boundary_flux = boundary_condition_slip_wall(u_inner, normal_direction,
+                                                     x, t, surface_flux_function,
+                                                     equations)
+    end
+
+    return boundary_flux
+end
+
+@inline function source_terms_gravity(u, x, t,
+                                      equations::CompressibleEulerPotentialTemperatureEquations2D)
+    rho, _, _, _ = u
+    return SVector(zero(eltype(u)), zero(eltype(u)), -equations.g * rho,
+                   zero(eltype(u)))
+end
+
+# Rotate momentum flux. The same as in compressible Euler.
+@inline function rotate_to_x(u, normal_vector::AbstractVector,
+                             equations::CompressibleEulerPotentialTemperatureEquations2D)
+    # cos and sin of the angle between the x-axis and the normalized normal_vector are
+    # the normalized vector's x and y coordinates respectively (see unit circle).
+    c = normal_vector[1]
+    s = normal_vector[2]
+
+    # Apply the 2D rotation matrix with normal and tangent directions of the form
+    # [ 1    0    0   0;
+    #   0   n_1  n_2  0;
+    #   0   t_1  t_2  0;
+    #   0    0    0   1 ]
+    # where t_1 = -n_2 and t_2 = n_1
+
+    return SVector(u[1],
+                   c * u[2] + s * u[3],
+                   -s * u[2] + c * u[3],
+                   u[4])
+end
+
+# Low Mach number approximate Riemann solver (LMARS) from
+# X. Chen, N. Andronova, B. Van Leer, J. E. Penner, J. P. Boyd, C. Jablonowski, S.
+# Lin, A Control-Volume Model of the Compressible Euler Equations with a Vertical Lagrangian
+# Coordinate Monthly Weather Review Vol. 141.7, pages 2526–2544, 2013,
+# https://journals.ametsoc.org/view/journals/mwre/141/7/mwr-d-12-00129.1.xml.
+@inline function flux_LMARS(u_ll, u_rr, normal_direction::AbstractVector,
+                            equations::CompressibleEulerPotentialTemperatureEquations2D)
+    @unpack a = equations
+    # Unpack left and right state
+    rho_ll, v1_ll, v2_ll, p_ll = cons2prim(u_ll, equations)
+    rho_rr, v1_rr, v2_rr, p_rr = cons2prim(u_rr, equations)
+
+    v_ll = v1_ll * normal_direction[1] + v2_ll * normal_direction[2]
+    v_rr = v1_rr * normal_direction[1] + v2_rr * normal_direction[2]
+
+    # Compute the necessary interface flux components
+    norm_ = norm(normal_direction)
+
+    rho = 0.5 * (rho_ll + rho_rr)
```

Comment:
In Trixi.jl, we are switching everything to `0.5f0` to enable `Float32` computations as well. Could you please do this here as well, e.g., 

```suggestion
    rho = 0.5f0 * (rho_ll + rho_rr)
```

See https://trixi-framework.github.io/Trixi.jl/stable/conventions/#Numeric-types-and-type-stability